Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -248,8 +247,21 @@ configure verbosity hcPath hcPkgPath conf0 = do
       compilerId :: CompilerId
       compilerId = CompilerId GHC ghcVersion
 
+      -- The @AbiTag@ is the @Project Unit Id@ but with redundant information from the compiler version removed.
```

Comment:
With this logic the ABITag should not be used by `Cabal` to reconstitute the unit-id, since the unit-id is not always the package version + the hash.

Reply:
On https://github.com/haskell/cabal/tree/wip/teo-precise-unit-id I'm getting this from "Project Unit Id"